hyperspy/signal.py

```diff
@@ -1924,8 +1924,18 @@ def _get_undefined_axes_list(self):
     def __call__(self, axes_manager=None):
         if axes_manager is None:
             axes_manager = self.axes_manager
-        return np.atleast_1d(
-            self.data.__getitem__(axes_manager._getitem_tuple))
+        if getattr(self._plot, 'pointer', None) is not None:
+            indices = self._plot.pointer.indices
+            size = self._plot.pointer.get_size_in_indices()
+            array_slices = [slice(None, None, None)] * \
+                self.axes_manager.navigation_dimension
+            for i, axis in enumerate(self.axes_manager.navigation_axes[::-1]):
```

Comment:
I mean the width of the pointer and therefore the slice in the corresponding axis (I think!). Yes, what your describe is more or less what I was thinking of, but I am not familiar enough with the details of the axes_manager and axis to know if this will have issue or if it does not fit the design of all this.
This is why I am asking about this!